drivers/adc/adc_stm32.c

```diff
@@ -948,11 +969,9 @@ static int start_read(const struct device *dev,
 
 	adc_context_start_read(&data->ctx, sequence);
 
-	err = adc_context_wait_for_completion(&data->ctx);
-
-	adc_stm32_teardown_channels(dev, index);
```

Comment:
no more adc_stm32_teardown_channels()  ?

Reply:
Not removed, it moved to `adc_context_on_complete`.